### ISSUE TITLE
Cyclic Graph Traversal is Incorrect for Child Mappings

### ISSUE THREAD

**Issue by blakewatters** [RestKit/RestKit#1142]
This is a follow-up issue from #1113 extracted from the comments of @waltflanagan:

I may have spoken too soon @blakewatters

I missed this issue since I was out for the holidays.

We're seeing a problem where our reply objects from above have another relationship added to them that is not cyclic. This mapping is used for content of replies (in practice objects containing data about audio/video content hosted on the web). Our mapping is constructed as follows:

RKEntityMapping\* replyMapping = [RKEntityMapping mappingForEntityForName:@"Reply" inManagedObjectStore:[RKManagedObjectStore defaultStore]];
      ... 
[replyMapping addRelationshipMappingWithSourceKeyPath:@"Replies" mapping:replyMapping];
[replyMapping addRelationshipMappingWithSourceKeyPath:@"Content" mapping:[MappingFactory contentMapping];

We're seeing an NSUnknownKeyException thrown inside RKAddObjectsInGraphWithCyclicKeyPathsToMutableSet at line 236: NSSet *objectsAtCyclicKeyPath = RKFlattenCollectionToSet([graph valueForKeyPath:cyclicKeyPath]);

The cyclicKeyPaths parameter passed into the method contains both 'Replies' and 'Content' but the current graph object in this conetxt is a Content object which does not have any cyclic relationships.

The mapping result in deleteLocalObjectsMissingFromMappingResult contains a mixed set of Content and Reply objects and those are iterated over and valueForKey'd for both keypaths that were assumed to be cyclic. This seems to be happening because the visition is marked as cyclic because Replies is a cyclic path, but the paths that are used come from mapping.relationshipMappings.destinationKeyPath.

Simply put, it looks like when a visitation is marked as cyclic, all relationships are then assumed to be cyclic when traversing the objects in the graph.

For now we're simply wrapping that line in a @try{} @catch{} but we'd like something a bit less messy to get around the issue.

Thanks again.
